types/documentdb/index.d.ts

```diff
@@ -126,12 +129,8 @@ interface AbstractMeta extends UniqueId {
 
 /** Represents a custom document for storing in DocumentDB  */
 interface NewDocument<TContent> extends UniqueId {
```

Comment:
Ah, okay. It might have been removed previously or just left there as information. In any case TContent is non-inferable from the context. So specifying the annotation for the document is really the caller's responsibility. So, yes, removing `doc` and `TContent` is probably best.